source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -413,15 +701,87 @@ CFDDEMSolver<dim>::check_contact_search_step_dynamic(const unsigned int &)
   return contact_detection_step;
 }
 
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::no_load_balance(const unsigned int &)
+{
+  return false;
+}
+
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::check_load_balance_once(const unsigned int &)
+{
+  bool load_balance_step = (this->simulation_control->get_step_number() ==
+                            this->cfd_dem_simulation_parameters.dem_parameters
+                              .model_parameters.load_balance_step);
+
+  if (load_balance_step || checkpoint_step)
+    {
+      load_balance();
+    }
+
+  return load_balance_step;
+}
+
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::check_load_balance_frequent(const unsigned int &)
+{
+  bool load_balance_step = (this->simulation_control->get_step_number() %
+                              this->cfd_dem_simulation_parameters.dem_parameters
+                                .model_parameters.load_balance_frequency ==
+                            0);
+
+  if (load_balance_step || checkpoint_step)
+    {
+      load_balance();
+    }
+
+  return load_balance_step;
+}
+
```

Comment:
I would move the check_load_balance_frequent outside of the solver (same for DEM) and instead of having it call load_balance, just return the boolean.
You can do the call to load_balance outside depending on the value of the boolean, it would make the code easier to understand

Reply:
done